src/components/SearchBar/SearchBarMobile.tsx

```diff
@@ -0,0 +1,253 @@
+'use client';
+
+import { Button } from '@components/ui/button';
+import {
+  DialogClose,
+  DialogContent,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+} from '@components/ui/dialog';
+import { Input } from '@components/ui/input';
+import { Popover, PopoverTrigger } from '@components/ui/popover';
+import { H2, Small } from '@components/ui/typography';
+import cn from '@lib/utils';
+import { Dialog, DialogTrigger } from '@radix-ui/react-dialog';
+import { motion } from 'framer-motion';
+import { HashIcon, LucideIcon, MapIcon, SearchIcon, XIcon } from 'lucide-react';
+import Image from 'next/image';
+import Link from 'next/link';
+import { useRouter } from 'next/navigation';
+import { useRef, useState } from 'react';
+import MenuItemContainer from './MenuItemContainer';
+import useDimensions from './use-dimensions';
+
+type SearchBarMobileProps = {
+  className: string;
+  activityPictures: Array<{
+    thumbnail: string;
+    url: string;
+    description: string;
+  }>;
+  activityKeywords: Array<{
+    url: string;
+    keyword: string;
+  }>;
+  locations: Array<{
+    url: string;
+    text: string;
+  }>;
+  categories: Array<{
+    icon: LucideIcon;
+    url: string;
+    text: string;
+  }>;
+  debugMode: boolean;
+};
+
+const locationMenu = {
+  open: (height = 1000) => ({
+    clipPath: `circle(${height * 2 + 200}px at 40px 800px)`,
+    transition: {
+      type: 'spring',
+      stiffness: 10,
+      restDelta: 2,
+    },
+  }),
+  closed: {
+    // TODO: use mouse event to get cursor position
+    clipPath: 'circle(0px at 40px 800px)',
+    transition: {
+      delay: 0.5,
+      type: 'spring',
+      stiffness: 400,
+      damping: 40,
+    },
+  },
+};
+
+const categoryMenu = {
+  open: (height = 1000) => ({
+    clipPath: `circle(${height * 2 + 200}px at 340px 800px)`,
+    transition: {
+      type: 'spring',
+      stiffness: 10,
+      restDelta: 2,
+    },
+  }),
+  closed: {
+    // TODO: use mouse event to get cursor position
+    clipPath: 'circle(0px at 340px 800px)',
+    transition: {
+      delay: 0.5,
+      type: 'spring',
+      stiffness: 400,
+      damping: 40,
+    },
+  },
+};
+
+const SearchBarMobile = ({
+  className,
+  activityPictures,
+  activityKeywords,
+  locations,
+  categories,
+  debugMode,
+}: SearchBarMobileProps) => {
+  const [isCategoryMenuOpen, setCategoryMenuOpen] = useState(false);
+  const [isLocationMenuOpen, setLocationMenuOpen] = useState(false);
+  const containerRef = useRef(null);
+  const { height } = useDimensions(containerRef);
+  const router = useRouter();
+
+  return (
+    <Dialog defaultOpen={debugMode}>
+      <DialogTrigger
+        className={cn(
+          'fixed bottom-0 right-0 z-10 flex flex-col items-center justify-center gap-4 bg-primary p-7 font-medium text-white',
+          className
+        )}
+      >
+        <SearchIcon className="size-6" />
+        <p>搜尋活動</p>
```

Comment:
標題下方增加 `margin-bottom` 與點擊「地區」、「類型」後，標題與底線的間距一致
ref: https://www.figma.com/design/6JPbEOj0WIGCAr3gJQU0Cb/%E5%85%AD%E8%A7%92_ChillKa?node-id=560-2627&t=vLc6P3UqjBDVezU2-0
```suggestion
        <p className="mb-1">搜尋活動</p>
```

Reply:
這按鈕應該是顯示在首頁的，應與對應的修改位置不同

figma:
https://www.figma.com/design/6JPbEOj0WIGCAr3gJQU0Cb/%E5%85%AD%E8%A7%92_ChillKa?node-id=500-1833&t=l3YX2UEQ3SSkvBd4-4

Reply:
好的